Core/src/Detector/IndexedRootVolumeFinderBuilder.cpp

```diff
@@ -0,0 +1,92 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Detector/IndexedRootVolumeFinderBuilder.hpp"
+
+#include "Acts/Detector/DetectorVolume.hpp"
+#include "Acts/Detector/detail/CylindricalDetectorHelper.hpp"
+#include "Acts/Detector/detail/GridAxisGenerators.hpp"
+#include "Acts/Navigation/DetectorVolumeFinders.hpp"
+#include "Acts/Utilities/Enumerate.hpp"
+
+namespace {
+
+template <typename Grid2D>
+void fillGridIndices2D(
```

Comment:
Is this also expected to be used elsewhere or could this become a private helper method of the ```IndexedRootVolumeFinderBuilder```?

Reply:
The functionality will be expanded for xy grids, etc when we adapt this to telescope geometries, for the moment, I would leave it conventiently.